### PR TITLE
RPL module

### PR DESCRIPTION
**Pull request by xkevin190**
### Contribution description

added an rpl implementation based on the functions used in the riot shell.
the function of this module is to start the rpl protocol to verify if it is a dodag node or a slave node.

### Testing procedure
 cd test/rpl

 make flash term

### CODE REVIEW

File: firmware/network/rpl_protocol/Makefile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+CFLAGS += -DGNRC_RPL_LIFETIME_UNIT=1 -DGNRC_RPL_DEFAULT_LIFETIME=32
+CFLAGS += -DGNRC_RPL_REGULAR_DAO_INTERVAL=13
+CFLAGS += -DGNRC_RPL_DEFAULT_DIO_INTERVAL_DOUBLINGS=13
+
+include $(RIOTBASE)/Makefile.base
```
Comment (eduazocar) [inline firmware/network/rpl_protocol/Makefile:6]: Same here

File: firmware/network/rpl_protocol/rpl_protocol.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    if (err != 0) {
+        printf("Error: couldn't init the radio module .\n");
+        return -1;
+    }
+
+    kernel_pid_t iface = get_ieee802154_iface();
+    if (iface == -1) {
+        printf("Error: could not get the iface.\n");
+        return -1;
+    }
```
Comment (eduazocar) [inline firmware/network/rpl_protocol/rpl_protocol.c:80]: This provides setup only to radio_devices

File: tests/rpl/main.c [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/*
```
Comment (eduazocar) [inline tests/rpl/main.c:1]: static-test empty_lines

File: tests/rpl/main.c [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+#include "embUnit.h"
+#include "uniqueid.h"
+#include "rpl_protocol.h"
+#include "radio.h"
+
+#define IFACE_ID 4
+#define DODAG_INSTANCE 1
+
+
```
Comment (eduazocar) [inline tests/rpl/main.c:33]: I think that also refers to these lines

**Review by eduazocar — CHANGES_REQUESTED** [review]
Ok. Fix a little formats errors

File: tests/rpl/main.c [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @brief       storage file
```
Comment (RocioRojas) [inline tests/rpl/main.c:18]: check the brief

**Review by eduazocar — APPROVED** [review]
Good Job :+1:

**Comment by eduazocar** [comment]
Go Ahead, Everything looks great :+1:

**Pull request merged**